src/schema.rs

```diff
@@ -421,10 +409,87 @@ impl Schema {
         }
     }
 
-    /// Parse a `serde_json::Value` representing a primitive Avro type into a
-    /// `Schema`.
-    fn parse_primitive(primitive: &str) -> AvroResult<Self> {
-        match primitive {
+    /// Create a `Schema` from a string representing a JSON Avro schema.
+    pub fn parse_str(input: &str) -> Result<Schema, Error> {
+        let mut parser = Parser::default();
+        parser.parse_str(input)
+    }
+
+    /// Create a array of `Schema`'s from a list of JSON Avro schemas. It is allowed that
```

Comment:
I have added to the doc string that this is only supported for name-able types, but I haven't added validation against it. I can if we want to do that. It will still work if you put in something like 
```json
{ "name": "A", "type": "array", "items": "double"}
```
even though arrays aren't name-able per the avro spec.  In fact I used that in the test cases, but I changed them all to actually name-able types just in case :slightly_smiling_face: